help_to_heat/frontdoor/views.py

```diff
@@ -411,13 +416,27 @@ def post(self, request, session_id, page_name, is_change_page=False):
             answers = interface.api.session.get_session(session_id)
             self.handle_saved_answers(request, session_id, page_name, answers, is_change_page)
 
+            next_page_name = get_next_page(page_name, answers)
+
             if is_change_page:
+                if next_page_name in schemas.change_page_override_pages:
+                    return redirect("frontdoor:change-page", session_id=session_id, page_name=next_page_name)
+
                 assert page_name in schemas.change_page_lookup
-                next_page_name = schemas.change_page_lookup[page_name]
-            else:
-                next_page_name = get_next_page(page_name, answers)
-                if next_page_name == unknown_page:
-                    return redirect("/sorry")
+                change_page = schemas.change_page_lookup[page_name]
+                start_of_journey = schemas.change_page_start_of_journey_lookup[change_page]
+
+                try:
+                    # ensure the user has answers to complete the journey
+                    calculate_journey(answers, from_page=start_of_journey, to_page=change_page)
+                    return redirect("frontdoor:page", session_id=session_id, page_name=change_page)
+                except CouldNotCalculateJourneyException as e:
+                    # if not, take them to the question that is preventing them from finishing
```

Comment:
I think some more explanation about why this error means they need to go to that page would be helpful.

Reply:
have clarified a bit